src/services/ban_service.py

```diff
@@ -1,20 +1,79 @@
 import discord
 import logging
+from typing import Optional, Tuple
 from util import log_info_and_embed, send_dm
+from settings import get_settings
 
+settings = get_settings()
 logger = logging.getLogger(__name__)
 
 
-async def ban_user(logging_embed: discord.Embed, user: discord.Member, reason: str):
-    log_info_and_embed(logging_embed, logger, f"going to ban {user.mention}")
+async def ban_user(
+    user: discord.Member, reason: str
+) -> Optional[Tuple[bool, bool, str]]:
+    """Executes ban of user
 
-    log_info_and_embed(logging_embed, logger, f"because {reason}")
+    Args:
+        user (discord.Member): mention or "@"/user ID of the user being banned.
+        reason (str): description of ban reason.
 
-    # TO DO: When the appeal process is implemented, add a link to the appeal process in the message.
-    await send_dm(
-        user,
-        f"You are being banned from NA Ultimate Raiding - FF XIV for the following reason: \n> {reason} \
-        \nYou may appeal this ban by contacting the moderators of the server in 30 days.",
-    )
+    Returns:
+        Optional[Tuple[bool, bool, str]]: tuple containing (result,dm_state,result_description)
+            - result (bool): Indicates whether the ban succeeded.
+            - dm_state (bool): Indicates whether the DM notification to the user succeeded.
+            - result_description (str): A message describing the status of the ban operation.
+    """
+    if len(reason) < 512:
+        result_description = ""
+        dm_state = False
+        # attempt to send a DM to the user with the reason for the ban
+        try:
+            await send_dm(
+                user,
+                f"You are being banned from NA Ultimate Raiding - FF XIV for the following reason: \n> {reason} "
+                "\nYou may appeal this ban by contacting the moderators of the server in 30 days.",
+            )
+            logger.info(f"Successfully sent dm to {user.mention}")
+            dm_state = True
+        except discord.Forbidden as e:
+            logger.error(f"DM to {user.mention} failed due to permission error: {e}")
+            dm_state = False
+        except discord.HTTPException as e:
+            logger.error(f"DM to {user.mention} due to HTTP Error: {e}")
+            dm_state = False
+        except Exception as e:
+            logger.error(f"DM to {user.mention} due to Unknown Error: {e}")
+            dm_state = False
```

Comment:
imo i feel like this amount of error handling is a bit extreme as it is cluttering up a lot of space than necessary. since we're already printing out the exception, and we're not particularly printing anything different/more useful than what the `Exception` object is doing, i think we can simplify to just use the catch-all `Exception`

Reply:
Fixed in latest version